### PR TITLE
tests: added tests for all commands except run, write, copy, and version

### PR DESCRIPTION
**Pull request by dgonyeo**
An issue was found where remove commands will not emit an error if asked to remove a nonexistent item. This commit fixes that.

Additionally this commit adds test for the following commands:
- annotation add
- annotation remove
- dependency add
- dependency remove
- end
- environment add
- environment remove
- label add
- label remove
- mount add
- mount remove
- port add
- port remove
- set-exec
- set-group
- set-name
- set-user

### CODE REVIEW

**Comment by dgonyeo** [comment]
Addresses most of https://github.com/appc/acbuild/issues/24

**Comment by jonboulle** [comment]
>  remove commands will not emit an error if asked to remove a nonexistent item. 

SHOULD we emit an error in this case? There might be something to be said for commands being (optionally?) idempotent?

**Comment by dgonyeo** [comment]
I think if you try to remove a mount point, and the mount point doesn't exist, it should at least say something. Perhaps just print a warning to stderr and have a 0 exit code?

**Comment by dgonyeo** [comment]
Honestly I'd just prefer to leave it on the user to ignore the error if they're removing something they're not sure would exist.

**Comment by jonboulle** [comment]
I am somewhat receptive to that but then it should be something consistent at the CLI layer (e.g. probably a particular error code) so that people can actually predictably ignore it.

/cc @krobertson thoughts?

File: test [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+set -e
+
+ORG_PATH="github.com/appc"
+REPO_PATH="${ORG_PATH}/acbuild"
+TESTS_PATH="${REPO_PATH}/tests"
+
+export ACBUILD_BIN="${PWD}/bin/acbuild"
+export GOPATH=${PWD}/gopath
```
Comment (jonboulle) [inline test]: Quote, for consistency

File: test [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+set -e
+
```
Comment (jonboulle) [inline test]: source ./build ?

Comment (dgonyeo) [inline test]: Sure, but it adds about 2 seconds on my laptop to running the tests. Probably for the best, so I don't confuse myself in the future as to why changes to `lib/` aren't being reflected in the tests.

**Comment by dgonyeo** [comment]
Should be ready for another pass. When acbuild is asked to remove something that doesn't exist the exit code will be 2, so that users can check what type of error was encountered. It'll also be easy going forward to add other error types with different exit codes.

File: lib/common.go [hunk 3]
```diff
@@ -25,6 +25,22 @@ import (
 
 const defaultWorkPath = ".acbuild"
 
+// ErrNotFound is returned when acbuild is asked to remove an element from
+// a list and the element is not present in the list
+var ErrNotFound = fmt.Errorf("element to be removed does not exist in this ACI")
+
+// GetErrorCode will return the recommended exit code for acbuild given an error
+func GetErrorCode(err error) int {
```
Comment (jonboulle) [inline lib/common.go]: Interesting approach. I think this function should exist in the CLI package instead though (but still key off the lib.Err types)

Comment (dgonyeo) [inline lib/common.go]: Yeah, I guess it does make more sense to be there. Fixed.

**Comment by dgonyeo** [comment]
Should be ready for another pass.

File: test (hunk 2 above)
Comment (krobertson) [inline test]: you can possibly do a `go test -i ./...` before `go test` itself.

**Comment by krobertson** [comment]
LGTM.  Oddly, github doesn't do syntax highlighting on the "set-user_test.go" files... maybe doesn't like the dash?

For the previous point about erroring when removing something not found, general command line behavior is to error.  If you do just `rm foo` and foo doesn't exist, it'll be an error. Maybe a `-f` flag would make sense in the future?

File: test (hunk 2 above)
Comment (dgonyeo) [inline test]: All of the dependencies for the tests should be vendored, so I don't think that would do anything.

**Pull request merged**